projects/ng-components/src/lib/controls/advanced-search/advanced-search.component.ts

```diff
@@ -0,0 +1,81 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { SearchOption, FilterClause, QueryParameters, Operators } from './models';
+
+@Component({
+  selector: 'lib-advanced-search',
+  templateUrl: './advanced-search.component.html'
+})
+export class AdvancedSearchComponent implements OnInit {
+  @Output() advancedSearchChanged: EventEmitter<FilterClause[]> = new EventEmitter<FilterClause[]>();
+  @Input('search-options') searchOptions: SearchOption[] = [];
+  @Input() filters?: FilterClause[];
+
+  public selectedField?: SearchOption;
+  public fieldValue?: string;
+  public fieldValueDateFrom: any;
+  public fieldValueDateTo: any;
+
+  constructor() { }
+
+  public ngOnInit(): void { }
+
+  public selectedFieldChanged() {
+    this.fieldValue = undefined;
+  }
+
+  public search() {
+    if (!this.filters) {
+      this.filters = [];
+    }
```

Comment:
Δεν το χρειάζεσαι γιατί δεν θα έρθει ποτέ undefined (αν διορθώσεις το filters: FilterClause[] = [];)
```suggestion
    if (!this.filters) {
      this.filters = [];
    }
```